### PR TITLE
:seedling:  add condition and reason definition in operator

### PR DESCRIPTION
**Pull request by zhiweiyin318**
## Summary
move the condition and reason definitions of operator here.
## Related issue(s)

Fixes #

### CODE REVIEW

File: operator/v1/types_clustermanager.go [hunk 1]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
 	// Items is a list of deployment configurations for registration and work distribution controllers.
 	Items []ClusterManager `json:"items"`
 }
+
+const (
+	// The types of ClusterManager condition status.
+	// ConditionClusterManagerApplied is the ClusterManager condition status which means all components have been applied on the hub.
+	ConditionClusterManagerApplied = "Applied"
+	// ConditionHubRegistrationDegraded is the ClusterManager condition status which means the registration is not ready to server on the hub.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]:
```suggestion
	// ConditionHubRegistrationDegraded is the ClusterManager condition status which means the registration is not ready to serve on the hub.
```

File: operator/v1/types_clustermanager.go [hunk 2]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
...
+	ConditionHubRegistrationDegraded = "HubRegistrationDegraded"
+	// ConditionHubPlacementDegraded is the ClusterManager condition status which means the placement is not ready to server on the hub.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]:
```suggestion
	// ConditionHubPlacementDegraded is the ClusterManager condition status which means the placement is not ready to serve on the hub.
```

File: operator/v1/types_clustermanager.go [hunk 3]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
...
+	ConditionHubPlacementDegraded = "HubPlacementDegraded"
+	// ConditionProgressing is the ClusterManager condition status which means all components are applying on the hub.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]: it mostly means the cluster manager is in upgrading phase, or not fully started yet.

File: operator/v1/types_clustermanager.go [hunk 4]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
...
+	// ReasonServiceAccountSyncFailed is the reason of the ConditionClusterManagerApplied condition to show failed to apply serviceAccounts.
+	ReasonServiceAccountSyncFailed = "ServiceAccountSyncFailed"
+	// ReasonClusterManagerCRDApplyFailed is the reason of the ConditionClusterManagerApplied condition to show failed to apply CRDs.
+	ReasonClusterManagerCRDApplyFailed = "CRDApplyFailed"
+	// ReasonWebhookApplyFailed is the reason of the ConditionClusterManagerApplied condition to show failed to apply webhooks.
+	ReasonWebhookApplyFailed = "WebhookApplyFailed"
+
+	// ReasonDeploymentRolling is the reason of the ConditionProgressing condition to show the deployed deployments are rolling.
+	ReasonDeploymentRolling = "ClusterManagerDeploymentRolling"
+	// ReasonUpToDate is the reason of the ConditionProgressing condition to  show he deployed deployments are up-to-date.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]:
```suggestion
	// ReasonUpToDate is the reason of the ConditionProgressing condition to show the deployed deployments are up-to-date.
```

File: operator/v1/types_clustermanager.go [hunk 5]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
...
+	ReasonUpToDate = "ClusterManagerUpToDate"
+
+	// ReasonStorageVersionMigrationFailed is the reason of the ConditionMigrationSucceeded condition to show failed to migrate storageVersion.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]:
```suggestion
	// ReasonStorageVersionMigrationFailed is the reason of the ConditionMigrationSucceeded condition to show migrate storageVersion failed.
```

File: operator/v1/types_clustermanager.go [hunk 6]
```diff
@@ -339,3 +339,54 @@ type ClusterManagerList struct {
...
+	ReasonStorageVersionMigrationFailed = "StorageVersionMigrationFailed"
+	// ReasonStorageVersionMigrationProcessing is the reason of the ConditionMigrationSucceeded condition to show storageVersion migration is not completed.
+	ReasonStorageVersionMigrationProcessing = "StorageVersionMigrationProcessing"
+	// ReasonStorageVersionMigrationSucceed is the reason of the ConditionMigrationSucceeded condition to show storageVersion migration is not succeed.
+	ReasonStorageVersionMigrationSucceed = "StorageVersionMigrationSucceed"
+
+	// ReasonGetRegistrationDeploymentFailed is the reason of the ConditionRegistrationDegraded condition to show failed get registration deployment.
```
Comment (qiujian16) [inline operator/v1/types_clustermanager.go]:
```suggestion
	// ReasonGetRegistrationDeploymentFailed is the reason of the ConditionRegistrationDegraded condition to show getting registration deployment failed.
```

File: operator/v1/types_klusterlet.go [hunk 7]
```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
...
+	// The types of klusterlet condition status.
+	// ConditionKlusterletApplied is the klusterlet condition status which means all components have been applied on the managed cluster.
+	ConditionKlusterletApplied = "Applied"
+	// ConditionReadyToApply is a klusterlet condition status which means it is ready to apply the resources on the managed cluster.
+	ConditionReadyToApply = "ReadyToApply"
+	// ConditionKlusterletAvailable is the klusterlet condition status which means all components are available and ready to serve.
+	ConditionKlusterletAvailable = "Available"
+	// ConditionHubConnectionDegraded is the klusterlet condition status which means the agent on the managed cluster cannot access the hub cluster.
+	ConditionHubConnectionDegraded = "HubConnectionDegraded"
+	// ConditionRegistrationDesiredDegraded is the klusterlet condition status which means the registration agent on the managed cluster is not ready to server.
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]:
```suggestion
	// ConditionRegistrationDesiredDegraded is the klusterlet condition status which means the registration agent on the managed cluster is not ready to serve.
```

File: operator/v1/types_klusterlet.go [hunk 8]
```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
...
+	ConditionRegistrationDesiredDegraded = "RegistrationDesiredDegraded"
+	// ConditionWorkDesiredDegraded is the klusterlet condition status which means the work agent on the managed cluster is not ready to server.
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]:
```suggestion
	// ConditionWorkDesiredDegraded is the klusterlet condition status which means the work agent on the managed cluster is not ready to serve.
```

File: operator/v1/types_klusterlet.go [hunk 9]
```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
...
+	// ReasonKlusterletApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply resources.
+	ReasonKlusterletApplyFailed = "KlusterletApplyFailed"
+	// ReasonKlusterletCRDApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply CRDs.
+	ReasonKlusterletCRDApplyFailed = "CRDApplyFailed"
+	// ReasonManagedClusterResourceApplyFailed is the reason of ConditionKlusterletApplied condition to failed to apply resources on managed cluster.
+	ReasonManagedClusterResourceApplyFailed = "ManagedClusterResourceApplyFailed"
+	// ReasonManagementClusterResourceApplyFailed is the reason of ConditionKlusterletApplied condition  failed to apply resources on management cluster.
+	ReasonManagementClusterResourceApplyFailed = "ManagementClusterResourceApplyFailed"
+
+	// ReasonKlusterletPrepareFailed is the reason of ConditionReadyToApply condition to show failed to prepare before apply/delete resources.
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]: what does prepare mean?

File: operator/v1/types_klusterlet.go [hunk 10]
```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
...
+	// ConditionHubConnectionDegraded is the klusterlet condition status which means the agent on the managed cluster cannot access the hub cluster.
+	ConditionHubConnectionDegraded = "HubConnectionDegraded"
+	// ConditionRegistrationDesiredDegraded is the klusterlet condition status which means the registration agent on the managed cluster is not ready to server.
+	ConditionRegistrationDesiredDegraded = "RegistrationDesiredDegraded"
+	// ConditionWorkDesiredDegraded is the klusterlet condition status which means the work agent on the managed cluster is not ready to server.
+	ConditionWorkDesiredDegraded = "WorkDesiredDegraded"
+
+	// ReasonKlusterletApplied is the reason of ConditionKlusterletApplied condition to show resources are applied.
+	ReasonKlusterletApplied = "KlusterletApplied"
+	// ReasonKlusterletApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply resources.
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]:
```suggestion
	// ReasonKlusterletApplyFailed is the reason of ConditionKlusterletApplied condition to show applying resources failed.
```

File: operator/v1/types_klusterlet.go [hunk 11]
```diff
@@ -244,3 +244,64 @@ type KlusterletList struct {
...
+	ReasonKlusterletApplyFailed = "KlusterletApplyFailed"
+	// ReasonKlusterletCRDApplyFailed is the reason of ConditionKlusterletApplied condition to show failed to apply CRDs.
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]:
```suggestion
	// ReasonKlusterletCRDApplyFailed is the reason of ConditionKlusterletApplied condition to show applying CRDs failed.
```

File: operator/v1/types_klusterlet.go (hunk 9 above)
Comment (zhiweiyin318) [reply operator/v1/types_klusterlet.go]: prepare means to get kubeconfig in external-managed-kubeconfig secret successfully 
changed to :
ReasonKlusterletPrepareFailed is the reason of ConditionReadyToApply condition to show it is failed to get the kubeConfig of managed cluster from the external-managed-kubeconfig secret in the hosted mode.

 ReasonKlusterletPrepared is the reason of ConditionReadyToApply condition to show the kubeConfig of managed cluster is validated from the external-managed-kubeconfig secret in the hosted mode.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**